newm-server/src/main/kotlin/io/newm/server/features/marketplace/model/SaleAmountRequest.kt

```diff
@@ -0,0 +1,15 @@
+package io.newm.server.features.marketplace.model
+
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class SaleAmountRequest(
+    val ownerAddress: String,
+    val bundlePolicyId: String,
+    val bundleAssetName: String,
+    val bundleAmount: Long,
+    val costPolicyId: String?, // Optional: if missing defaults to NEWM token
+    val costAssetName: String?, // Optional: if missing defaults to NEWM token
```

Comment:
The default if missing is NEWM token, and that's the way that the frontend should use it for now. USD is not supported. But the API supports any Native Token.

Reply:
The smart contract was written with flexibility in mind since it is a decentralized system. Someone could list their tokens in the contact selling for SickCity (SICK) tokens if they wanted to. However, OUR frontend can be designed to ignore anything that isn't denominated in NEWM. That doesn't prevent somebody else from using our smart contracts for their own purposes.